src/dataflow/src/sink/tail.rs

```diff
@@ -171,18 +181,22 @@ fn tail<G>(
                     &mut packer,
                     connector.object_columns + 1,
                 );
-                if connector.emit_progress {
+                if connector.emit_progress && tail_protocol.is_some() {
                     if let Some(progress_row) = progress_row {
                         results.push(progress_row);
                     }
                 }
             }
 
-            // TODO(benesch): the lack of backpressure here can result in
```

Comment:
I'm guessing we don't want to keep this TODO because there's too many in the codebase anyways?

Reply:
Yeah, I felt weird deleting other folks' TODOs, but I think this has been safely copied to the point in the code where there might now be a problem (the coordinator, when it returns the data to the client). Should be safe to delete. @benesch just in case.

Reply:
Go for it! The TODO has been satisfied by hiring someone (@aljoscha) with structured thoughts about back pressure, IMO.